src/zen/folders/ZenFolders.mjs

```diff
@@ -103,11 +129,56 @@
   customElements.define('zen-folder', ZenFolder);
 
   class ZenFolders extends ZenPreloadedFeature {
+    #popup = null;
+    #popupTimer = null;
+    #mouseTimer = null;
+
     init() {
       this.#initContextMenu();
+      this.#initTabsPopup();
       this.#initEventListeners();
     }
 
+    #initContextMenu() {
+      const contextMenuItems = window.MozXULElement.parseXULToFragment(`
+          <menuitem id="zen-context-menu-new-folder" data-l10n-id="zen-toolbar-context-new-folder"/>
+          `);
+      document.getElementById('context_moveTabToGroup').before(contextMenuItems);
+    }
+
+    #initTabsPopup() {
+      const popupSet = document.querySelector('#mainPopupSet');
+      if (popupSet.querySelector('#zen-folder-tabs-popup')) return;
+      const popup = window.MozXULElement.parseXULToFragment(
+        `
+        <panel id="zen-folder-tabs-popup" type="arrow" orient="vertical">
```

Comment:
Instead of adding this dynamically, it can simply be added into https://github.com/zen-browser/desktop/tree/dev/src/browser/base/content/zen-panels for example